src/lib/components/ui/input.svelte

```diff
@@ -1,396 +1,402 @@
 <script lang="ts">
+	import type { Snippet } from 'svelte'
 	import type { HTMLInputAttributes } from 'svelte/elements'
+	import { WarningAltFilled, Information, Subtract, Add } from 'carbon-icons-svelte'
 	type Layout = 'vertical' | 'horizontal'
 	type Dimension = 'default' | 'large' | 'compact' | 'small'
 	interface Props extends HTMLInputAttributes {
+		label?: string
 		labelFor?: string
 		dimension?: Dimension
 		layout?: Layout
 		unit?: string
+		error?: Snippet
+		hover?: boolean
+		active?: boolean
+		focus?: boolean
+		controls?: boolean
 	}
 	let {
+		label,
 		labelFor = Math.random().toString(16),
 		placeholder,
 		value = $bindable(),
 		dimension = 'default',
 		layout = 'vertical',
 		unit,
+		error,
+		hover,
+		active,
+		focus,
+		controls,
 		type,
 		class: className = '',
+		children,
 		...restProps
 	}: Props = $props()
+	let inputValue: number = $state(value)
 </script>
 
-{#if layout == 'vertical'}
-	<div class="root {layout} {dimension} {className}">
-		<input id={labelFor} bind:value {placeholder} {type} {...restProps} />
-		<label class="label" for={labelFor}>
+<div class="root {layout} {dimension} {className}" class:controls>
+	<label class="label" for={labelFor}>
+		{label}
+	</label>
+	{#if children && layout === 'horizontal' && layout !== 'horizontal'}
+		<div class="helper-button">
+			<Information size={dimension === 'small' ? 16 : 24} />
+		</div>
+	{/if}
+	<div class="wrapper">
+		<input
+			id={labelFor}
+			class:active
+			class:hover
+			class:focus
+			class:error
+			bind:value={inputValue}
 			{placeholder}
-		</label>
-		{#if unit}
+			{type}
+			{...restProps}
+		/>
+		{#if unit && !error}
 			<label class="unit" for={labelFor}>{unit}</label>
 		{/if}
-		<div class="helper-text">
-			<slot />
-		</div>
-	</div>
-{:else}
-	<div class="root {layout} {dimension} {className}">
-		<input id={labelFor} bind:value {placeholder} {type} {...restProps} />
-		<label class="label" for={labelFor}>
-			<slot />
-		</label>
-		{#if unit}
-			<label for={labelFor} class="unit">{unit}</label>
+		{#if error}
+			<label class="error-icon" for={labelFor}>
+				<WarningAltFilled size={dimension === 'small' ? 16 : 24} />
+			</label>
+		{/if}
+		{#if controls && type === 'number'}
+			<div class="control-buttons">
+				<button class="substract" onclick={() => (inputValue -= 1)}
+					><Subtract size={dimension === 'small' ? 16 : 24} /></button
+				>
+				<button class="add" onclick={() => (inputValue += 1)}
+					><Add size={dimension === 'small' ? 16 : 24} /></button
+				>
```

Comment:
I believe these should be secondary buttons, should have a focus state and that focus state should remove the focus state from the input.